### PR TITLE
treat 'N/A' as nil if :na_as_nil option if true

### PR DESCRIPTION
**Pull request by ironsand**
I think it would be nicer if there is option that treats 'N/A' as `nil`.

thanks.

### CODE REVIEW

**Comment by herval** [comment]
can you write a test for this? tks!

**Comment by ironsand** [comment]
I wrote it, but I'm not good at testing. please tell me if I'm doing something wrong.

File: lib/yahoo-finance.rb [hunk 1]
```diff
@@ -117,6 +117,9 @@ def quotes(symbols_array, columns_array = [
       ret = []
       symbols_array.each_slice(SYMBOLS_PER_REQUEST) do |symbols|
         read_quotes(symbols.join("+"), columns_array).map do |row|
+          if options[:na_as_nil]
+            row.each { |item| item[-1] = nil if item[-1] == 'N/A' }
```
Comment (herval) [inline lib/yahoo-finance.rb]: why the `item[-1]`? Do you want this to always handle just the *last* column data? If so, why?

Comment (ironsand) [inline lib/yahoo-finance.rb]: `item` is like `[name, 'google`]. So I'm checking the value field.

Comment (herval) [inline lib/yahoo-finance.rb]: The `columns_array` is an arbitrary array, so it's not garanteed that name will be the last element:

```
> YahooFinance::Client.new.quotes(["BVSP", "NATU3.SA", "USDJPY=X"], [:ask, :bid, :last_trade_date])
 => [#<OpenStruct ask="N/A", bid="N/A", last_trade_date="N/A">, #<OpenStruct ask="N/A", bid="N/A", last_trade_date="10/13/2015">, #<OpenStruct ask="119.5530", bid="119.5510", last_trade_date="10/14/2015">]
```

You might want to check all rows for N/A, not just the last

Comment (ironsand) [inline lib/yahoo-finance.rb]: yeah, I'm checking all rows by `row.each`.
```
YahooFinance::Client.new.quotes(["BVSP", "NATU3.SA", "USDJPY=X"], [:ask, :bid, :last_trade_date], na_as_nil: true)
=> [#<OpenStruct ask=nil, bid=nil, last_trade_date=nil>,
 #<OpenStruct ask=nil, bid=nil, last_trade_date="10/19/2015">,
 #<OpenStruct ask="119.5200", bid="119.5175", last_trade_date="10/20/2015">]
```

**Comment by ironsand** [comment]
Hi, I updated the code more readable. how about this time?

**Comment by herval** [comment]
👍

**Pull request merged**